app/scripts/controllers/product/CreateRateController.js

```diff
@@ -0,0 +1,62 @@
+/**
+ * Created by Jose on 24/07/2017.
+ */
+(function (module) {
+    mifosX.controllers = _.extend(module, {
+        CreateRateController: function (scope, resourceFactory, location, dateFilter, translate, webStorage) {
+            scope.template = [];
+            scope.formData = {};
+            scope.first = {};
+            scope.rateError = false;
+            scope.translate = translate;
+
+
+            // resourceFactory.chargeTemplateResource.get(function (data) {
+            //     scope.template = data;
+            //     scope.showChargePaymentByField = true;
+            //     scope.chargeCalculationTypeOptions = data.chargeCalculationTypeOptions;
+            //     scope.chargeTimeTypeOptions = data.chargeTimeTypeOptions;
+            //
+            //     scope.incomeAccountOptions = data.incomeOrLiabilityAccountOptions.incomeAccountOptions || [];
+            //     scope.liabilityAccountOptions = data.incomeOrLiabilityAccountOptions.liabilityAccountOptions || [];
+            //     scope.incomeAndLiabilityAccountOptions = scope.incomeAccountOptions.concat(scope.liabilityAccountOptions);
+            // });
+
+            //Right now only loan is accepted for a rate.
+            scope.rateOptions = [{id : "m_loan"}];
+
+
+            scope.rateAppliesToSelected = function (rateAppliesId) {
```

Comment:
what is the purpose of this function

Reply:
Honestly it doesn't seems to do something important. Someone else worked in the code before me so my best bet is that it was used just for some tests